nixos/modules/services/web-apps/immich.nix

```diff
@@ -91,7 +91,7 @@ in
     };
     port = mkOption {
       type = types.port;
-      default = 3001;
```

Comment:
why change the default port? Seems it could complicate peoples configs if they never set the port option. I would prefer if this option was kept at the current value.

Reply:
It's described in the second commit:

> This was always upstream's default but they also change the internal port, i.e. behind the reverse proxy, to 2283 in https://github.com/immich-app/immich/pull/13185.

This could be a breaking change though. Should this be behind a stateVersion flag?

Reply:
> Should this be behind a stateVersion flag?

I thought about that but I don't think it's necessary since the module hasn't made it to a stable NixOS version yet.

Reply:
> > Should this be behind a stateVersion flag?
> 
> I thought about that but I don't think it's necessary since the module hasn't made it to a stable NixOS version yet.

People are still using the unstable branch though. In the end the port number doesn't really matter.

Reply:
> This could be a breaking change though. Should this be behind a stateVersion flag?

It's too heavy for something not released yet and immich clearly advertises they wont let backward-compatibility in their way. 
This worked for me after updating the reverse proxy. Phone app and CLI worked as well.

Reply:
After some thinking I agree with @dotlambda on this topic.

Reply:
My config is
```nix
services.nginx = {
    enable = true;
    virtualHosts."immich.example.com" = {
      enableACME = true;
      forceSSL = true;
      locations."/" = {
        proxyPass = "http://${config.services.immich.host}:${toString config.services.immich.port}";
        proxyWebsockets = true;
      };
      extraConfig = ''
        client_max_body_size 512M;
      '';
    };
  };
```
so I don't even need to change anything.

Reply:
I agree that a stateVersion change would likely be too heavy. But I do wonder what the benefit of changing the port at all is. The upstream port choice really has no bearing on what we use in NixOS so I would have just left what was originally assigned.

That being said I also used `config.services.immich.port` in my config so it won't be a bother for me. But in general I would avoid changing things that don't need changing.